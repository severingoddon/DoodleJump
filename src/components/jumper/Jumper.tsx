import React from 'react';
import './jumper-css.css';

let classnameJumper = 'doodle-jumper';
let classnameJumperStore = 'doodle-jumper';

export const setThemeJumper = (isDark: boolean) => {
    if(isDark) {
        classnameJumper = 'doodle-jumper-dark';
        classnameJumperStore = 'doodle-jumper-dark';
    }
    else {
        classnameJumper = 'doodle-jumper';
        classnameJumperStore = 'doodle-jumper';
    }
};

export const setRocket = (rocketOn: boolean) => {
    if(rocketOn) {
        if(classnameJumperStore === 'doodle-jumper') {
            classnameJumper = 'doodle-jumper-rocket';
        }
        else if(classnameJumperStore === 'doodle-jumper-dark') {
            classnameJumper = 'doodle-jumper-rocket-dark';
        }
    }else classnameJumper = classnameJumperStore;
};

export default (props: any) => {

    const style = {
        left: `${props.doodlerCoords[0]}px`,
        bottom: `${props.doodlerCoords[1]}px`,
        transform:`scaleX(${props.image})`
    };

    return(
        <div data-testid="jumper" className={classnameJumper} style={style}/>
    );
};