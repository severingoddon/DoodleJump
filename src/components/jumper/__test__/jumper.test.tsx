import React from 'react';
import ReactDOM from 'react-dom';
import Jumper from '../Jumper';
import {render, cleanup} from '@testing-library/react';
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);

const doodlerCoords = [200, 150];

it('renders jumper correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Jumper image="1" doodlerCoords={doodlerCoords}/>, div);
});

it('sets the css attribute bottom correctly', () => {
    const {getByTestId} = render(<Jumper image="1" doodlerCoords={doodlerCoords}/>);
    expect(getByTestId('jumper').style.bottom).toBe(doodlerCoords[1] + 'px');
});

it('sets the css attribute left correctly', () => {
    const {getByTestId} = render(<Jumper image="1" doodlerCoords={doodlerCoords}/>);
    expect(getByTestId('jumper').style.left).toBe(doodlerCoords[0] + 'px');
});

it('sets the css attribute scaleX(1) correctly', () => {
    const {getByTestId} = render(<Jumper image="1" doodlerCoords={doodlerCoords}/>);
    expect(getByTestId('jumper').style.transform).toBe('scaleX(1)');
});

it('sets the css attribute scaleX(-1) correctly', () => {
    const {getByTestId} = render(<Jumper image="-1" doodlerCoords={doodlerCoords}/>);
    expect(getByTestId('jumper').style.transform).toBe('scaleX(-1)');
});

it('matches snapshot', () => {
    const tree = TestRenderer.create(<Jumper image="1" doodlerCoords={doodlerCoords}/>).toJSON();
    expect(tree).toMatchSnapshot();
});



