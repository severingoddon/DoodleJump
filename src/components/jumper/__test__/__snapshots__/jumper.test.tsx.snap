// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="doodle-jumper"
  data-testid="jumper"
  style={
    Object {
      "bottom": "150px",
      "left": "200px",
      "transform": "scaleX(1)",
    }
  }
/>
`;
